File: app/src/main/java/network/devandroid/com/networklibrarycomparison/BaseActivity.java
package network.devandroid.com.networklibrarycomparison;

import android.content.pm.ActivityInfo;
import android.os.Bundle;
import android.support.annotation.LayoutRes;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.View;

import butterknife.ButterKnife;
import butterknife.Unbinder;

public abstract class BaseActivity extends AppCompatActivity {

    private Unbinder mUnbinder;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayoutResource());
        mUnbinder = ButterKnife.bind(this);

        //setup toolbar
        final Toolbar toolBar = getToolBar();
        if (toolBar != null) {
            setSupportActionBar(toolBar);
            toolBar.setNavigationIcon(R.drawable.ic_nav_back);
            toolBar.setNavigationOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    //default call backpress
                    onBackPressed();
                }
            });
        }

        //force orientation to portrait
        setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mUnbinder.unbind();
    }

    @LayoutRes
    protected abstract int getLayoutResource();

    protected abstract Toolbar getToolBar();
}

File: app/src/main/java/network/devandroid/com/networklibrarycomparison/BaseActivity.kt
package network.devandroid.com.networklibrarycomparison

import android.content.pm.ActivityInfo
import android.os.Bundle
import android.support.annotation.LayoutRes
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar

import net.hockeyapp.android.CrashManager
import net.hockeyapp.android.UpdateManager

import butterknife.ButterKnife
import butterknife.Unbinder

abstract class BaseActivity : AppCompatActivity() {

    private var unbinder: Unbinder? = null

    @get:LayoutRes
    protected abstract val layoutResource: Int

    protected abstract val getToolBar: Toolbar

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(layoutResource)
        unbinder = ButterKnife.bind(this)

        //setup toolbar
        if (getToolBar != null) {
            setSupportActionBar(getToolBar)
            getToolBar.setNavigationIcon(R.drawable.ic_nav_back)
            getToolBar.setNavigationOnClickListener {
                //default call backpress
                onBackPressed()
            }
        }

        //force orientation to portrait
        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT

        //check for hockey app updates
        checkForUpdates()
    }

    override fun onResume() {
        super.onResume()
        // ... your own onResume implementation
        checkForCrashes()
    }

    override fun onDestroy() {
        super.onDestroy()
        unbinder!!.unbind()
        unregisterManagers()
    }

    private fun checkForUpdates() {
        // Remove this for store builds!
        UpdateManager.register(this)
    }

    private fun unregisterManagers() {
        UpdateManager.unregister()
    }

    private fun checkForCrashes() {
        CrashManager.register(this)
    }
}
